### ISSUE TITLE
componentDidUpdate is triggered twice while using ConnectedRouter

### ISSUE THREAD

**Issue by scssyworks** [supasate/connected-react-router#129]
While switching routes as depicted below, my component assigned to this route is rendered twice. As a result ``componentDidUpdate`` is executed two times.

``/route/to/page1`` --> ``/route/to/page2`` (route: ``/route/to/:page``)

**Comment by shae-wat** [comment]
Im seeing the same behavior, is the intention to have devs implement logic in componentDidUpdate to determine whether their components actually need to update? Is there a way that ConnectedRouter could give us that behavior/logic out of the box?

**Comment by vorlov** [comment]
the same behavior is observed when I click back on browser button

**Comment by benj3** [comment]
The same is also observed with forward and backwards buttons on browser.

**Comment by n-sviridenko** [comment]
Having a similar issue. It happens when calling `history.push()` or `dispatch(push())`. `<Switch />` renders twice. This causes: mounting -> unmounting -> mounting of the current route.

**Comment by gregnb** [comment]
Dealing with this same issue where my component will: mount, unmount and then mount again. Is there any solution to this problem?

**Comment by gregnb** [comment]
I found that changing any routes from using `component` prop to `render` prop did the trick:

Works
```
        <Switch>
          <Route path={'login'} render={props => <LoginContainer {...props} />} />
        </Switch>
```

Renders 2x
```
        <Switch>
          <Route path={'login'} component={props => <LoginContainer {...props} />} />
        </Switch>
```

**Comment by supasate** [comment]
It should be fixed in [v6.1.0](https://github.com/supasate/connected-react-router/releases/tag/v6.1.0). Please let me know if it helps and feel free to re-open this issue if the problem still exists.
